app/gui2/src/components/CodeEditor.vue

```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
+import { ref, watchEffect } from 'vue'
+import {useWindowEvent} from "@/util/events";
+
+const emit = defineEmits<{
+  codeUpdate: [code: string]
+}>()
+
+const content = ref('main = \n    2 + 2')
+const shown = ref(false)
+
+watchEffect(() => emit('codeUpdate', content.value))
+
+useWindowEvent('keydown', e => { if (e.key == `\``) shown.value = !shown.value})
```

Comment:
This should be addressed now. But whatever key we choose, cannot be types within the code editor, as otherwise we cannot close it while it is focused. Since that is what we did in the old GUI, I think that is the behaviour we should keep for now.